homeassistant/helpers/state.py

```diff
@@ -197,10 +197,11 @@ def async_handle_service_calls(coro_list):
         yield from asyncio.wait(execute_tasks, loop=hass.loop)
 
 
-def state_as_number(state):
+def state_as_number(state, mapping=None):
```

Comment:
I don't see a reason to make a function for 3 lines. This function was initialized to handle the state types that are the same over hole hass. But you mapping a customize, so it will not usefully for the core. So please move it into your component and other component can add this 3 lines and maybe make more dynamic for his platform.